src/script.js

```diff
@@ -251,54 +331,90 @@ function isMultisigInput (script, allowIncomplete) {
   return chunks.slice(1).every(isCanonicalSignature)
 }
 
-function isMultisigOutput (script) {
-  var chunks = decompile(script)
-  if (chunks.length < 4) return false
-  if (chunks[chunks.length - 1] !== OPS.OP_CHECKMULTISIG) return false
-
-  var mOp = chunks[0]
-  var nOp = chunks[chunks.length - 2]
+function decodeMultisigOutput (scriptPubKey, chunks) {
+  if (chunks.length < 4) {
+    throw new Error('Multisig script should contain at least 4 elements')
+  }
+  if (chunks[chunks.length - 1] !== OPS.OP_CHECKMULTISIG) {
+    throw new Error('Final opcode should be OP_CHECKMULTISIG')
+  }
+  if (!types.Number(chunks[0])) {
+    throw new Error('First element must be a number-push opcode')
+  }
+  if (!types.Number(chunks[chunks.length - 2])) {
+    throw new Error('Second-last element must be a number-push opcode')
+  }
+  var m = chunks[0] - OP_INT_BASE
+  var n = chunks[chunks.length - 2] - OP_INT_BASE
 
-  if (!types.Number(mOp)) return false
-  if (!types.Number(nOp)) return false
+  if (m <= 0) {
+    throw new Error('Number of signatures required must be greater than zero')
+  }
+  if (n > 16) {
+    throw new Error('Number of public keys must be less than 17')
+  }
+  if (m > n) {
+    throw new Error('Number of signatures cannot exceed the number of public keys')
+  }
+  if (n !== chunks.length - 3) {
+    throw new Error('n does not match the number of public keys')
+  }
 
-  var m = mOp - OP_INT_BASE
-  var n = nOp - OP_INT_BASE
+  var keys = chunks.slice(1, -2)
+  if (!keys.every(isCanonicalPubKey)) {
+    throw new Error('Non-canonical pubic key found')
+  }
 
-  // 0 < m <= n <= 16
-  if (m <= 0) return false
-  if (m > n) return false
-  if (n > 16) return false
-  if (n !== chunks.length - 3) return false
+  return {
+    type: 'multisig',
+    nRequiredSigs: m,
+    nPublicKeys: n,
+    publicKeyBuffers: keys
+  }
```

Comment:
This should return the same arguments that were used to `encode` it.

That is,

`function multisigOutput (m, pubKeys) {`

So, lets just return `{ m: m,  pubKeys: pubKeys }`